lib/v2/telegram/router.js

```diff
@@ -1,5 +1,20 @@
+const config = require('@/config').value;
+const core_router = require('@/core_router');
+const tglibChannel = require('./tglib/channel');
+const webChannel = require('./channel');
+
 module.exports = function (router) {
-    router.get('/channel/:username/:routeParams?', require('./channel'));
+    if (config.telegram.session) {
+        // core_router does not cache, which is necessary for video streaming
+        core_router.get('/telegram/channel/:username/:media(.+)', tglibChannel.getMedia);
+    }
+
+    router.get('/channel/:username/:routeParams?', (ctx) => {
+        // tglib impl does not support routeParams yet
+        const useWeb = ctx.params.routeParams || !config.telegram.session;
+        return useWeb ? webChannel(ctx) : tglibChannel(ctx);
+    });
```

Comment:
```suggestion
    if (config.telegram.session) {
        // core_router does not cache, which is necessary for video streaming
        coreRouter.get('/telegram/channel/:username/:media(.+)', require('./tglib/channel').getMedia);
    }

    router.get('/channel/:username/:routeParams?', (ctx) => {
        // tglib impl does not support routeParams yet
        const useWeb = ctx.params.routeParams || !config.telegram.session;
        return useWeb ? require('./channel')(ctx) : require('./tglib/channel')(ctx);
    });
```

In this way, gramjs won't [start with RSSHub](https://github.com/DIYgod/RSSHub/actions/runs/7031510001/job/19133265093#step:12:12)